lib/mail_room/structured_logging/structured_logger.rb

```diff
@@ -0,0 +1,26 @@
+require 'logger'
+require 'json'
+
+module MailRoom
+  module StructuredLogging
+    class StructuredLogger < Logger
+
+      def format_message(severity, timestamp, progname, message)
+        raise ArgumentError.new("Message must be a Hash") unless message.is_a? Hash
+
+        data = {}
+        data[:severity] = severity
+        data[:time] = timestamp || Time.now.to_s
+        # only accept a Hash
+        data.merge!(message)
+
+        data.to_json + "\n"
+      end
+
+      def noop?
```

Comment:
Where does this get called?

Reply:
```suggestion
      def noop?
```

Reply:
Actually, looking at the internals of `Logger`, if `@logdev` is nil, it is a noop already. We don't need to add any special code.

Sorry for making you do that. This can come out.

Reply:
I think I'll merge this, and then I can just remove this in another commit.